app/helpers/application_helper/toolbar_builder.rb

```diff
@@ -796,7 +796,6 @@ def build_toolbar_hide_button(id)
       when "miq_template_refresh"
         return true if @record && !@record.ext_management_system && !(@record.host && @record.host.vmm_product.downcase == "workstation")
       when "miq_template_scan", "image_scan"
-        return true unless @record.is_available?(:smartstate_analysis) || @record.is_available_now_error_message(:smartstate_analysis)
```

Comment:
Made relevant changes as suggested by @h-kataria . Also added rspec tests for new code changes.